azure/application-gateway/advanced/variables.tf

```diff
@@ -369,4 +369,119 @@ variable "waf_configuration" {
     })), {})
   })
   default = null
+}
+
+variable "listener_waf_configuration" {
+  description = "Defining the WAF policy per listener"
+  type = map(object({ // Key = policy name
+    associated_listeners = list(string)
+    firewall_mode        = optional(string, "Prevention")
```

Comment:
should this not be detection by default?

Reply:
Most LLD's request prevention was the thinking here